lib/client.js

```diff
@@ -751,6 +771,7 @@
         this._heartbeat = setTimeout(() => {
 
             this.onError(new NesError('Disconnecting due to heartbeat timeout', errorTypes.TIMEOUT));
+            this._onDisconnect();
```

Comment:
Then again, it may just have to be considered a breaking change even if it does provide a `log`? It should probably include some meta information that this is not a "real" disconnect, but a "will disconnect soon"?

Reply:
As we're the ones closing the connection we could fake this log object here. Or we could instead have a `willDisconnect()` handler and keep `onDisconnect()` for server received disconnect only?

Reply:
It doesn't make a lot of difference to us... But I'm not sure how other people use it?

willDisconnect sounds backwards compat, though, at the cost of some extra complexity...